src/dune_rules/dune_file.ml

```diff
@@ -1006,9 +1006,56 @@ end
 
 module Install_conf = struct
   module File_entry = struct
+    module Without_include = struct
+      type t =
+        | File_binding of File_binding.Unexpanded.t
+        | Glob_files of Glob_files.t
+
+      let decode =
+        let open Dune_lang.Decoder in
+        let file_binding_decode =
+          let+ file_binding = File_binding.Unexpanded.decode in
+          File_binding file_binding
+        in
+        let glob_files_decode =
+          let version_check = Dune_lang.Syntax.since Stanza.syntax (3, 5) in
+          let+ glob_files =
+            sum
+              [ ( "glob_files"
+                , version_check >>> String_with_vars.decode >>| fun glob ->
+                  Glob_files.{ glob; recursive = false } )
+              ; ( "glob_files_rec"
+                , version_check >>> String_with_vars.decode >>| fun glob ->
+                  Glob_files.{ glob; recursive = true } )
+              ]
+          in
+          Glob_files glob_files
+        in
+        file_binding_decode <|> glob_files_decode
+
+      let to_file_bindings_unexpanded t ~expand_str =
+        let open Memo.O in
+        match t with
+        | File_binding file_binding -> Memo.return [ file_binding ]
+        | Glob_files glob_files ->
+          let+ paths =
+            Glob_files.expand_memo glob_files ~f:(fun sw ->
+                expand_str sw >>| Path.of_string)
+          in
+          let glob_loc = String_with_vars.loc glob_files.glob in
+          List.map paths ~f:(fun path ->
+              let src = (glob_loc, Path.to_string path) in
+              File_binding.Unexpanded.make ~src ~dst:src)
+
+      let to_file_bindings_expanded t ~expand_str ~dir =
+        let open Memo.O in
+        to_file_bindings_unexpanded t ~expand_str
+        >>= Memo.List.map ~f:(File_binding.Unexpanded.expand ~dir ~f:expand_str)
```

Comment:
It's a matter of style, but you can just do `Memo.bind` and remove the `let open` uses only once.